apps/gateway/lib/gateway/controllers/fallback_controller.ex

```diff
@@ -64,6 +64,10 @@ defmodule Gateway.Controllers.FallbackController do
     send_resp(conn, 400, Jason.encode!(%{"error" => "the user doesn't have that quest"}))
   end
 
+  def call(conn, {:error, _failed_operation, :not_found, _changes_so_far}) do
+    send_resp(conn, 404, Jason.encode!(%{"error" => "not found"}))
+  end
```

Comment:
This is ok for the specific case when user is not found, but the with clause has many other operations. Are all of them properly handled?
They were being caught by a wild card in the with-else statement.

Reply:
fair point, i added a default case